selfdrive/ui/navd/route_parser.cc

```diff
@@ -0,0 +1,1228 @@
+
+    if (laneInstruction.isEmpty())
+    {
+        if (wayName.isEmpty())
+            return MapboxRouteParser::tr("Continue straight");
+        else
+            return MapboxRouteParser::tr("Continue straight onto %1").arg(wayName);
+    }
+
+    switch (direction)
+    {
+    case QGeoManeuver::DirectionForward:
+        if (wayName.isEmpty())
+            //: This string will be prepended with lane instructions. E.g., "Use the left or the right lane and continue straight"
+            return laneInstruction + MapboxRouteParser::tr(" and continue straight");
+        else
+            return laneInstruction + MapboxRouteParser::tr(" and continue straight onto %1").arg(wayName);
+    case QGeoManeuver::DirectionHardLeft:
+        if (wayName.isEmpty())
+            return laneInstruction + MapboxRouteParser::tr(" and make a sharp left");
+        else
+            return laneInstruction + MapboxRouteParser::tr(" and make a sharp left onto %1").arg(wayName);
+    case QGeoManeuver::DirectionLeft:
+        if (wayName.isEmpty())
+            return laneInstruction + MapboxRouteParser::tr(" and turn left");
+        else
+            return laneInstruction + MapboxRouteParser::tr(" and turn left onto %1").arg(wayName);
+    case QGeoManeuver::DirectionLightLeft:
+    case QGeoManeuver::DirectionBearLeft:
+        if (wayName.isEmpty())
+            return laneInstruction + MapboxRouteParser::tr(" and make a slight left");
+        else
+            return laneInstruction + MapboxRouteParser::tr(" and make a slight left onto %1").arg(wayName);
+    case QGeoManeuver::DirectionHardRight:
+        if (wayName.isEmpty())
+            return laneInstruction + MapboxRouteParser::tr(" and make a sharp right");
+        else
+            return laneInstruction + MapboxRouteParser::tr(" and make a sharp right onto %1").arg(wayName);
+    case QGeoManeuver::DirectionRight:
+        if (wayName.isEmpty())
+            return laneInstruction + MapboxRouteParser::tr(" and turn right");
+        else
+            return laneInstruction + MapboxRouteParser::tr(" and turn right onto %1").arg(wayName);
+    case QGeoManeuver::DirectionLightRight:
+    case QGeoManeuver::DirectionBearRight:
+        if (wayName.isEmpty())
+            return laneInstruction + MapboxRouteParser::tr(" and make a slight right");
+        else
+            return laneInstruction + MapboxRouteParser::tr(" and make a slight right onto %1").arg(wayName);
+    case QGeoManeuver::DirectionUTurnLeft:
+    case QGeoManeuver::DirectionUTurnRight:
+        if (wayName.isEmpty())
+            return laneInstruction + MapboxRouteParser::tr(" and make a U-turn");
+        else
+            return laneInstruction + MapboxRouteParser::tr(" and make a U-turn onto %1").arg(wayName);
+    default:
+        return laneInstruction;
+    }
+}
+
+static QString instructionText(const QJsonObject &step, const QJsonObject &maneuver, QGeoManeuver::InstructionDirection direction)
```

Comment:
You're aware we don't use the instructionText anywhere? We only use the banner objects from the response to decide what to show to the user.

Reply:
gotcha, thanks